server/ecore-glsp/src/main/java/com/eclipsesource/glsp/ecore/operationhandler/EcoreDeleteOperationHandler.java

```diff
@@ -25,10 +26,19 @@
 public class EcoreDeleteOperationHandler extends DeleteOperationHandler {
 	@Override
 	protected boolean delete(String elementId, GModelIndex index, GraphicalModelState graphicalModelState) {
+		
 		super.delete(elementId, index, graphicalModelState);
 		EcoreModelState modelState = EcoreModelState.getModelState(graphicalModelState);
+		
+		modelState.getIndex().getSemantic(elementId).ifPresent(element -> {
+			if(element instanceof EReference && ((EReference) element).getEOpposite() != null) {
+				EcoreUtil.remove(((EReference) element).getEOpposite());
```

Comment:
In general, EcoreUtil.delete() should be preferred, as it will not only remove the object from the model, but also clean all references to that object (Preventing DanglingHref exceptions)